File: DSA_PYTHON/Search/LinearSearch.py
class LinearSearch:

    def linear(self, array, target):
        for index, value in enumerate(array):
            if value == target:
                return index
        return -1

    def main(self):
        print("------------------------------------")
        array = [6, 2, 7, 4, 8, 10]

        print("Enter element to search: ")
        n = int(input())
        res = self.linear(array, n)
        if res == -1:
            print("Item not found")
        else:
            print(f"Item found at index {res}")
        return res

File: DSA_CSHARP/Search/LinearSearch.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DSA_CSHARP.Search;
internal class LinearSearch
{
    public int Linear(int[] array, int n)
    {

        for (int i = 0; i < n; i++)
        {
            if (array[i] == n)
                return i;
        }
        return -1;
    }
    public string Main()
    {
        Console.WriteLine($"------------------------------------");
        var array = new int[] { 6, 2, 7, 4, 8, 10 };

        Console.Write($"Enter element to search: ");
        var n = Convert.ToInt32(Console.ReadLine());
        var res = Linear(array, n);

        if (res == -1)
            Console.WriteLine($"Item not found");
        else
            Console.WriteLine($"Item found at index {res}");


        return res.ToString();
    }
}
